### PR TITLE
fix(store,world): minor config validation fixes

### PR DESCRIPTION
**Pull request by alvrs**
Minor fixes to config input validations:
- `systems.openAccess` incorrectly expected `true` as the only valid input. It now allows `boolean`.
- The config complained if parts of it were defined `as const` outside the config input. This is now possible.
- Shorthand inputs are now enabled.

### CODE REVIEW

File: .changeset/dry-toes-lay.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+"@latticexyz/store": patch
+"@latticexyz/world": patch
+---
+
+Minor fixes to validations of the new config:
```
Comment (alvrs) [inline .changeset/dry-toes-lay.md:6]:
```suggestion
Minor fixes to validations of the config:
```

Comment (holic) [inline .changeset/dry-toes-lay.md:6]:
```suggestion
Minor fixes to TS validation of the config:
```

Comment (alvrs) [inline .changeset/dry-toes-lay.md:6]:
```suggestion
Minor fixes to config input validations:
```

File: packages/store/ts/config/v2/store.test.ts [hunk 2]
```diff
@@ -495,4 +495,12 @@ describe("defineStore", () => {
...
     ).throwsAndHasTypeError("Overrides of `name` and `namespace` are not allowed for tables in a store config");
   });
+
+  it("should allow const enum as input", () => {
+    const enums = {
+      Example: ["First", "Second"],
+    } as const;
+
+    attest(defineStore({ enums }).enums).equals(enums);
+  });
```
Comment (holic) [inline packages/store/ts/config/v2/store.test.ts:505]: cc @0xhank

**Review by holic — DISMISSED** [review]

**Pull request merged**